docs/ocean/design_docs/min_level_cell.rst

```diff
@@ -0,0 +1,450 @@
+changes.
+
+Implementation: No significant loss of performance
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2021/02/11
+
+Contributors: Xylar Asay-Davis
+
+We will take care not to introduce unnecessary ``if`` statements or equivalents
+that were not present before.
+
+
+Testing
+-------
+
+Testing: Inactive top cells
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2021/02/11
+
+Contributors: Xylar Asay-Davis
+
+We will ensure that all 3D, prognostic variables have been initialized to the
+NetCDF fill value in inactive cells at the top of the water column.  We will
+attempt to do the same for 3D diagnostic variables.  This should expose any
+computations involving these invalid cells.
+
+We will modify the ``sub_ice_shelf_2D`` configuration in legacy COMPASS to
+support a z-level initial coordinate in the ice-shelf cavity (including writing
+out ``minLevelCell`` in the initial condition).  Then, we will run the
+following test cases in this configuration with the new coordinate:
+
+* ``default``
+
+* ``restart_test``
+
+* ``iterative_init``
+
+We will run on multiple machines (Ubuntu laptop, Anvil, Grizzly, Compy) with a
+mix of Gnu and Intel compilers.  We will plot the resulting T, S and KE fields
+to make sure they look comparable to the results with the current
+terrain-following coordinate.
+
+We will test with both the linear (I believe the default) and JM equations of
+state.
+
+Testing: Surface fields and fluxes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2021/02/11
+
+Contributors: Xylar Asay-Davis
+
+The testing in ``sub_ice_shelf_2D`` will account for the surface fluxes and
+fields that we most anticipate being affected by ``minLevel*``.  Testing of
+other surface fields will likely require running tests that include GM, KPP
+and other parameterizations that are not part of this configuration by default.
+
+Testing: Inactivate cells during simulation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2021/02/11
+
+Contributors: Xylar Asay-Davis
+
+Since this is just a consideration, no testing is necessary.  We will add
+timers around ``ocn_init_routines_compute_min_max_level()`` (if they are not
+already present) to see if the timing is significant, though a global test
+may be necessary to get a realistic feel.
+
+Testing: Bit-for-bit when no inactive top cells
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2021/02/11
+
+Contributors: Xylar Asay-Davis
+
+We will run the legacy COMPASS nightly and land-ice-fluxes regression suites
+on multiple machines (Ubuntu laptop, Anvil, Grizzly, Compy) with a mix of Gnu
+and Intel compilers to make sure nothing has changed when we run with
+``minLevelCell = 1`` everywhere.
+
+We will also run the following E3SM tests before and after the changes:
+
+* ``SMS.T62_oQU120_ais20.MPAS_LISIO_TEST.anvil_intel``
+* ``<<help>>``
```

Comment:
@mattdturner, @mark-petersen and @philipwjones, I could use recommendations on other E3SM tests to make sure we include.  We need pretty good code coverage to make sure we don't break anything.

Reply:
I would like to see testing with all production resolutions and comp sets as well.  I don't know the exact test setup lines, pinging @jonbob too for his input on tests.

Reply:
@vanroekel and @jonbob, could you help us make a list of tests we need to run?  We'd like to do this testing this week and put in a PR.